lib/the_little_thinkers_space_web/controllers/profile_controller.ex

```diff
@@ -3,41 +3,55 @@ defmodule TheLittleThinkersSpaceWeb.ProfileController do
 
   alias TheLittleThinkersSpace.Accounts
   alias TheLittleThinkersSpace.Accounts.Profile
+  alias TheLittleThinkersSpace.Accounts.User
 
-  def index(conn, _params) do
-    profiles = Accounts.list_profiles()
-    render(conn, "index.html", profiles: profiles)
+  plug :authenticate_user when action in [:index, :new, :create, :show, :edit, :update, :delete]
+  action_fallback TheLittleThinkersSpaceWeb.FallbackController
+
+  def index(conn, _params, current_user) do
+    with :ok <- Bodyguard.permit(Profile, :index, current_user) do
+      profiles = Accounts.list_profiles()
+      render(conn, "index.html", profiles: profiles)
+    end
   end
 
-  def new(conn, _params) do
+  def new(conn, _params, _current_user) do
     changeset = Accounts.change_profile(%Profile{})
     render(conn, "new.html", changeset: changeset)
   end
 
-  def create(conn, %{"profile" => profile_params}) do
-    case Accounts.create_profile(profile_params) do
+  def create(conn, %{"profile" => profile_params}, %User{id: id}) do
+    profile_params
+    |> Map.put("user_id", id)
+    |> Accounts.create_profile()
+    |> case do
       {:ok, profile} ->
         conn
         |> put_flash(:info, "Profile created successfully.")
         |> redirect(to: Routes.profile_path(conn, :show, profile))
 
+      {:error, %Ecto.Changeset{errors: [user_id: _]}} ->
+        conn
+        |> put_flash(:error, "User already has a profile")
```

Comment:
You are addressing the person who is making their own profile right?

```suggestion
        |> put_flash(:error, "You already have a profile")
```